src/tracer/MOM_generic_tracer.F90

```diff
@@ -572,6 +614,16 @@ subroutine MOM_generic_tracer_column_physics(h_old, h_new, ea, eb, fluxes, Hml,
     enddo ; enddo
     sosga = global_area_mean(surface_field, G, unscale=US%S_to_ppt)
 
+    if (CS%mld_pha_fixed) then
+      mld_pha(:,:) = CS%mld_pha_val
+    else
+      if (CS%mld_pha_use_delta_rho) then
+        call diagnoseMLDbyDensityDifference(-1, h_old, tv, CS%mld_pha_drho, G, GV, US, CS%diag, ref_h_mld=CS%mld_pha_href, MLD_out=mld_pha)
```

Comment:
This needs to be updated:

`        call diagnoseMLDbyDensityDifference(-1, h_old, tv, CS%mld_pha_drho, G, GV, US, CS%diag, CS%mld_pha_href, -1, -1, MLD_out=mld_pha)`

Also `MLD_out`is msssing in current `diagnoseMLDbyDensityDifference` and `diagnoseMLDbyEnergy` subroutines.

Reply:
This should be added to the diagnose MLD routines with this PR.